File: leetcode_java/19. Remove Nth Node From End of List.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode front = head;
        for(int i = 0; i < n; i++)
            front = front.next;
        if(front == null) return head.next;
        ListNode rear = head;
        while(front.next != null) {
            front = front.next;
            rear = rear.next;
        }
        rear.next = rear.next.next;
        return head;
    }
}

File: leetcode_python/19. Remove Nth Node From End of List.py
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        front = head
        for _ in range(n):
            front = front.next
        if not front: return head.next
        rear = head
        while front.next:
            front = front.next
            rear = rear.next
        rear.next = rear.next.next
        return head


# no leetcode solution